scripts/common.js

```diff
@@ -647,8 +647,185 @@ export function extractTopicsAndProducts() {
+      // in order to show the Region name either from SessionStorage language or blog.language in our region button
+      if (regionLang !== window.blog.language) {
+        // if the selected Region lang do not match the blog.language we get the Region name based on the sessionStorage saved Language Value
+        const storedLanguage = regionsNameList.find(r => r.lang === regionLang);
+        regionName = storedLanguage.name;
+      } else {
+        // else we get the Region name base on the blog.language and after have find the same language in our RegionListName
+        if (langPage !== undefined) {
+            regionName = langPage.name;
+        }       
+      }
+    } else {
+      regionLang = regionPage.lang;
+      regionName = regionPage.name;
+      // sessionStorage will be used only if current Region lang is not same as blog.lang as en_apac or en_uk 
+      // where blog.lang will still being en
+      if (regionPage.lang !== window.blog.language) {
+        sessionStorage.setItem('blog-selected-language', regionPage.lang);
+      }
+    } 
+    return {regionLang, regionName};
+  }
+
+  const dropdownRegionList = document.querySelector('.region-dropdown-list');
+  const {regionLang, regionName} = getSelectedRegion(); 
+
+  // Change Region name value from Feds Region Picker Button adding the actual Region Name
+  const FEDSregionPickerText = document.querySelector('.feds-regionPicker-text');
+  if (FEDSregionPickerText && regionName !== undefined) {
+      FEDSregionPickerText.innerText = regionName;
+  }
+
+  // Automatically build the dropdown based on Region List
+  if (dropdownRegionList) {
+    for (const {lang, name, home} of regionsNameList) {
+      dropdownRegionList.insertAdjacentHTML('afterbegin', `<li><a class="region-dropdown-picker" tabindex="0" href="${home}" title="${name}" data-lang="${lang}">${name}</a></li>`);
+      const regionDropdownPicker = document.querySelector('.region-dropdown-picker');
+      if (regionDropdownPicker) {
+        // Mark the actual selected Region on the dropdown
+        if (regionLang === lang) {
+          regionDropdownPicker.classList.add('selected');
+        }
+      }
+    }
+  }
+
+  const regionDropdownButton =  document.querySelector('.feds-regionPicker');
+  if (regionDropdownButton) {
+    regionDropdownButton.addEventListener('click', (event) => {
+      event.preventDefault();
+      toggleDropdownModal();
+    });
+  }
+
+  // Hide region modal if clicked outside
+  document.addEventListener('click', function (event) {
+    if (regionDropdownButton || HTMLElement) {
+      if (!event.target.closest('.region-dropdown') && !event.target.closest('.feds-regionPicker')) {
+        hideDropdownModal();
+      }
+    }
+  });
+
+  // Keyboard access for Region Picker
+  if (regionDropdownButton) {
+    document.body.addEventListener('keyup', (event) => {
+      if (event.key === 'Escape') {
+          toggleDropdownModal();
```

Comment:
If the Region Picker is not visible, pressing `Escape` will make it visible. I guess you need to use `hideDropdownModal()` to ensure it gets hidden.